packages/component-meta/scripts/generateComponentMeta.ts

```diff
@@ -221,9 +221,31 @@ const generateComponentMeta = (): void => {
     }
 
     let [, rawPropTypes] = /const [a-z][a-zA-z]+: (?:Omit<)?PropTypes<.+?> = ({[\s\S]+?});/.exec(source) || [];
+
+    // handle shared props
+    let sharedPropsExportFile; // also needed for import paths of variables
+    if (rawPropTypes && rawPropTypes.match(/\.{3}[a-zA-Z]+,/g)) {
+      const [, sharedPropsConstName] = rawPropTypes.match(/(shared[a-zA-z]+Types)/);
```

Comment:
Why use a different regex than in the check above? Also why do we need to check first and then extract in a 2nd step? We could just try extract right from the beginning and then check if we extracted something.  
Also the scenario for multiple shared props isn't supported right now. This should easier be commented or supported.